kong/konnect_application_service_test.go

```diff
@@ -0,0 +1,105 @@
+package kong
+
+import (
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestKonnectApplicationService_Create(T *testing.T) {
+	RunWhenDBMode(T, "postgres")
+	RunWhenEnterprise(T, ">=3.6.0", RequiredFeatures{})
+
+	assert := assert.New(T)
+	require := require.New(T)
+
+	client, err := NewTestClient(nil, nil)
+	assert.NoError(err)
+	assert.NotNil(client)
+
+	var (
+		clientID        = uuid.NewString()
+		consumerGroup   = []string{uuid.NewString()}
+		scopes          = []string{"/auth"}
+		authStrategyID  = uuid.NewString()
+		exhaustedScopes = []string{"/eauth"}
+		orgID           = uuid.NewString()
+		developerID     = uuid.NewString()
+		createdAt       = time.Now().Unix()
+	)
+
+	kaa := &KonnectApplication{
+		ID:              &clientID,
+		ClientID:        clientID,
+		ConsumerGroups:  consumerGroup,
+		Scopes:          scopes,
+		AuthStrategyID:  &authStrategyID,
+		ExhaustedScopes: exhaustedScopes,
+		ApplicationContext: &ApplicationContext{
+			OrganizationID: &orgID,
+			DeveloperID:    &developerID,
+		},
+		CreatedAt: createdAt,
+	}
+	createResponse, err := client.KonnectApplication.Create(defaultCtx, kaa)
+	require.NoError(err)
+	require.NotNil(createResponse)
+	require.Equal(createResponse.ClientID, clientID)
+	require.Equal(createResponse.CreatedAt, createdAt)
+	require.Equal(createResponse.ConsumerGroups, consumerGroup)
+	require.Equal(createResponse.Scopes, scopes)
+	require.Equal(createResponse.ExhaustedScopes, exhaustedScopes)
+
+	err = client.KonnectApplication.Delete(defaultCtx, createResponse.ID)
+	require.NoError(err)
```

Comment:
Just to be sure that the entity gets deleted when one of the assertions fail

```suggestion
	createResponse, err := client.KonnectApplication.Create(defaultCtx, kaa)
	require.NoError(err)
	require.NotNil(createResponse)
	
	t.Cleanup(func() {
		assert.NoError(client.KonnectApplication.Delete(context.Background(), createResponse.ID))
	})
	
	require.Equal(createResponse.ClientID, clientID)
	require.Equal(createResponse.CreatedAt, createdAt)
	require.Equal(createResponse.ConsumerGroups, consumerGroup)
	require.Equal(createResponse.Scopes, scopes)
	require.Equal(createResponse.ExhaustedScopes, exhaustedScopes)
```